packages/flutter_tools/lib/src/ios/lldb.dart

```diff
@@ -0,0 +1,336 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// @docImport '../xcode_project.dart';
+library;
+
+import 'dart:async';
+
+import '../base/io.dart';
+import '../base/logger.dart';
+import '../base/process.dart';
+import '../convert.dart';
+
+/// LLDB is the default debugger in Xcode on macOS. Once the application has
+/// launched on a physical iOS device, you can attach to it using LLDB.
+///
+/// See `xcrun devicectl device process launch --help` for more information.
+class LLDB {
+  LLDB({required Logger logger, required ProcessUtils processUtils})
+    : _logger = logger,
+      _processUtils = processUtils;
+
+  final Logger _logger;
+  final ProcessUtils _processUtils;
+
+  _LLDBProcess? _lldbProcess;
+
+  /// Whether or not a LLDB process is running.
+  bool get isRunning => _lldbProcess != null;
+
+  /// The process id of the application running on the iOS device.
+  int? get appProcessId => _lldbProcess?.appProcessId;
+
+  _LLDBLogPatternCompleter? _logCompleter;
+
+  /// Pattern of lldb log when the process is stopped.
+  ///
+  /// Example: (lldb) Process 6152 stopped
+  static final _lldbProcessStopped = RegExp(r'Process \d* stopped');
+
+  /// Pattern of lldb log when the process is resuming.
+  ///
+  /// Example: (lldb) Process 6152 resuming
+  static final _lldbProcessResuming = RegExp(r'Process \d+ resuming');
+
+  /// Pattern of lldb log when the breakpoint is added.
+  ///
+  /// Example: Breakpoint 1: no locations (pending).
+  static final _breakpointPattern = RegExp(r'Breakpoint (\d+)*:');
+
+  /// Breakpoint script required for JIT on iOS.
+  ///
+  /// This should match the "handle_new_rx_page" function in [IosProject._lldbPythonHelperTemplate].
+  static const _pythonScript = '''
+"""Intercept NOTIFY_DEBUGGER_ABOUT_RX_PAGES and touch the pages."""
+base = frame.register["x0"].GetValueAsAddress()
+page_len = frame.register["x1"].GetValueAsUnsigned()
+
+# Note: NOTIFY_DEBUGGER_ABOUT_RX_PAGES will check contents of the
+# first page to see if handled it correctly. This makes diagnosing
+# misconfiguration (e.g. missing breakpoint) easier.
+data = bytearray(page_len)
+data[0:8] = b'IHELPED!'
+
+error = lldb.SBError()
+frame.GetThread().GetProcess().WriteMemory(base, data, error)
+if not error.Success():
+    print(f'Failed to write into {base}[+{page_len}]', error)
+    return
+
+# If the returned value is False, that tells LLDB not to stop at the breakpoint
+return False
+''';
+
+  /// Starts an LLDB process and inputs commands to start debugging the [appProcessId].
+  /// This will start a debugserver on the device, which is required for JIT.
+  Future<bool> attachAndStart(String deviceId, int appProcessId) async {
+    Timer? timer;
+    try {
+      timer = Timer(const Duration(minutes: 1), () {
```

Comment:
Use `startSpinner` and `slowWarningCallback`

Reply:
`startSpinner` doesn't work if there is already a spinner going: 
https://github.com/flutter/flutter/blob/3be04414e907eaf3cb56a4ce6f16b2045ee1f64d/packages/flutter_tools/lib/src/ios/devices.dart#L518